src/components/NotesMindMap/NotesMindMapReducer.js

```diff
@@ -49,59 +48,65 @@ export const notesMindMapReducer = (state = defaultState, { type, data }) => {
 
   const handleNoteNameUpdateRequestSuccessAction = () => {
     const updatedNote = data;
-    const newState = cloneTreeInState();
-    const noteToUpdate = tree(newState.rootNote).find(note => note.id === updatedNote.id);
-    noteToUpdate.name = updatedNote.name;
-    return newState;
+    let notes = [ ...state.notes ]
+    notes = notes.filter(note => note.id !== updatedNote.id)
+    notes.push({ id: updatedNote.id, label: updatedNote.name, name: updatedNote.name, isNote: updatedNote.isNote });
```

Comment:
`{ id: updatedNote.id, label: updatedNote.name, name: updatedNote.name, isNote: updatedNote.isNote }` is duplicated 2 times.

And also you have `{ id: newNote.id, name: newNote.name, label: newNote.name, isNote: newNote.isNote`, which I'd say duplicates the same thing for the 3rd time.

Might make sense to extract it into a function?